src/AggregateFunctions/AggregateFunctionSumMap.h

```diff
@@ -115,7 +115,14 @@ class AggregateFunctionMapBase : public IAggregateFunctionDataHelper<
                         "Values for {} are expected to be Numeric, Float or Decimal, passed type {}",
                         getName(), value_type->getName()};
 
-                result_type = value_type_without_nullable->promoteNumericType();
+                WhichDataType value_type_to_check(value_type);
+
+                /// Do not promote decimal because of implementation issues of this function design
```

Comment:
Which issues? This comment doesn't really tell anything.

Reply:
There is a comment at the top of the function amateur quality, low perfomance, because it operates on column Fields. So when I saw that it does not work with Decimals there was 2 ways:
1. Totally rewrite function.
2. Do not promote to decimal type.
After discussion with Alexey we decided to choose 2 way, also for decimals we use same approach in other places. Probably comment can be removed.